opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -380,11 +395,11 @@ static inline int mca_btl_base_am_rdma_advance(mca_btl_base_module_t *btl,
     }
 
     mca_btl_base_descriptor_t *descriptor = context->descriptor;
-    mca_btl_base_rdma_hdr_t *hdr = (mca_btl_base_rdma_hdr_t *) descriptor->des_segments[0]
+    am_rdma_hdr_t *hdr = (am_rdma_hdr_t *) descriptor->des_segments[0]
                                        .seg_addr.pval;
-    const size_t packet_size = mca_btl_base_rdma_operation_size(btl, hdr->type, remaining);
+    const size_t packet_size = am_rdma_operation_size(am_module, hdr->type, remaining);
 
-    if (!mca_btl_base_rdma_is_atomic(hdr->type)) {
+    if (!am_rdma_is_atomic(hdr->type)) {
         hdr->data.rdma.size = packet_size;
         hdr->data.rdma.initiator_address = (uint64_t) context->local_address + context->sent;
     } else {
```

Comment:
unrelated to this PR, on line 406 it says: `hdr->data.atomic.size = packet_size;`, but packet_size is of type `size_t` and atomic.size is of type `uint8_t`.  Can you take a look at the size fields in `am_rdma_hdr_t`?

Reply:
Is this because this field will only ever be either `4` or `8` based on `flags & MCA_BTL_ATOMIC_FLAG_32BIT`?

Reply:
yes, but added a comment and an assert, because that's a hell of an assumption.